src/index.js

```diff
@@ -0,0 +1,178 @@
+}
+export function renderTasks(selectedId) {
+  selectedId.tasks.forEach((task) => {
+    const todoList = document.importNode(todoTemplate.content, true);
+    const card = todoList.querySelector('.card');
+    const todoTitle = todoList.querySelector('.card-title');
+    todoTitle.innerText = task.title;
+    const todoDescription = todoList.querySelector('.card-description');
+    todoDescription.innerText = task.description;
+    const todoPriority = todoList.querySelector('.card-priority');
+    todoPriority.innerText = task.priority;
+    todoPriority.classList.add('btn');
+    const todoDdate = todoList.querySelector('.card-footer');
+    todoDdate.innerText = `Due on ${task.ddate}`;
+    const editBtn = todoList.querySelector('.edit-btn');
+    const index = selectedId.tasks.indexOf(task);
+
+
+    editBtn.addEventListener('click', () => editTodo(task, index));
+    const deleteTaskBtn = todoList.querySelector('.delete-btn');
+    deleteTaskBtn.addEventListener('click', () => {
+      deleteTask(todoindex, card);
+    });
+    cards.appendChild(todoList);
+    addClass();
+  });
+}
+
+
+export function renderProjects() {
+  clearList(projects);
+
+  lists.forEach((list) => {
+    const project = document.createElement('li');
+    project.dataset.listId = list.id;
+    project.classList.add('list-group-item');
+    project.setAttribute('role', 'button');
+    project.innerText = list.name;
+    if (list.id === object.selectedId) {
+      project.classList.add('active');
+    }
+    projects.appendChild(project);
+  });
+}
+export function render() {
+  renderProjects();
+  const selectedProject = lists.find((list) => list.id === object.selectedId);
+  if (selectedProject == null) {
+    cards.style.display = 'none';
+  } else {
+    cards.style.style = '';
+    clearList(cards);
+    renderTasks(selectedProject);
+  }
+}
+
+
+render();
+function saveAndRender() {
+  render();
+  save();
+}
+function defaultProject() {
+  if (lists.length === 0) {
+    const project = createProject('Default');
+    object.selectedId = project.id;
+    lists.push(project);
+    saveAndRender();
+  }
+}
+
+defaultProject();
+
+newProjectForm.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const newProjectName = newProjectInput.value;
+  if (newProjectName == null || newProjectName === '') return;
+  const project = createProject(newProjectName);
+  newProjectInput.value = null;
+  lists.push(project);
+  saveAndRender();
+});
+
+projects.addEventListener('click', (e) => {
+  e.preventDefault();
+  if (e.target.tagName.toLowerCase() === 'li') {
+    object.selectedId = e.target.dataset.listId;
+    saveAndRender();
+  }
+});
+
+
+submit.addEventListener('click', (e) => {
+  e.preventDefault();
+  if (submit.classList.contains('edit')) {
+    const editindex = todoindex.value;
+    const selectedProject1 = lists.find((list) => list.id === object.selectedId);
+    const todo = selectedProject1.tasks[editindex];
+    todo.title = title.value;
```

Comment:
- Am unable to update a todo item. This is the error shown in the console
```javascript
Uncaught TypeError: i is undefined
<anonymous> index.js:140
```

Reply:
The edit and delete features are working as they are intended to work. And we do not have any variable " i"-- Uncaught TypeError: i is undefined. -- in our code.